### DISCUSSION TITLE
How to find out which version of the dag code created the run/task?

### DISCUSSION THREAD

**Discussion by slawomir-kluz-openx** [apache/airflow#25150] [Q&A]
Hi! Let us consider the following scenario:
- we have a working DAG version v1 which creates runes with multiple tasks
- we upgrade the DAG code to version v2

Is there a way to check which version of the DAG scheduled specific run/task?

**Comment by potiuk** [comment]
Nope. It is not possible. We need to implement DAG Versioning Improvement Proposal in order to have support for DAG Versioning https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-36+DAG+Versioning. For now the best practice is to change dag_id when you change it and want to keep track of history.
